src/atoms/Auth.ts

```
@@ -1,13 +1,79 @@
 export const featureFlagAtom = atom<{
   isMapVisible: boolean;
   isToiletVisible: boolean;
 } | null>(null);
+
+// useMe hook - centralized userInfo management
+export function useMe() {
+  const {api} = useAppComponents();
+  const [userInfo, _setUserInfoAtom] = useAtom(userInfoAtom);
+  const setFeatureFlag = useSetAtom(featureFlagAtom);
+
+  const _syncUserInfo = async (userInfo: User) => {
```

ESLint

[error] 'userInfo' is already declared in the upper scope on line 35 column 10.

(@typescript-eslint/no-shadow)